fcrepo-clients/src/main/java/edu/unc/lib/dl/event/RepositoryPremisLogger.java

```diff
@@ -133,6 +137,8 @@ public PremisLogger writeEvents(Resource... eventResources) {
             log.debug("Adding events to PREMIS log for {}", objPid);
 
             PID logPid = getMdEventsPid(objPid);
+
+            Lock logLock = lockManager.awaitWriteLock(logPid);
```

Comment:
At some point we should also add in read locking, so that it won't try to read while a write is being performed. I've created a followup ticket for doing this with the premis logger: https://jira.lib.unc.edu/browse/BXC-2722